src/metrics/cognitive.rs

```diff
@@ -1696,4 +1733,197 @@ mod tests {
+        );
+    }
+
+    #[test]
+    fn java_compound_conditions() {
+        check_metrics::<JavaParser>(
+            "class X {
+              public static void print(boolean a, boolean b, boolean c, boolean d){  
+                if(a && b){ // +2
+                  System.out.println(\"test1\");
+                }
+                if(c && d){ // +2
+                  System.out.println(\"test2\");
+                }
+              }
+            }",
+            "foo.java",
+            |metric| {
+                insta::assert_json_snapshot!(
+                    metric.cognitive,
+                    @r###"
+                    {
+                      "sum": 4.0,
+                      "average": 4.0,
+                      "min": 0.0,
+                      "max": 4.0
+                    }"###
+                );
+            },
+        );
+    }
+
+    #[test]
+    fn java_switch_statement() {
+        check_metrics::<JavaParser>(
+            "class X {
+              public static void print(boolean a, boolean b, boolean c, boolean d){
+                switch(expr){ //+1
+                  case 1:
+                    System.out.println(\"test1\");
+                    break;
+                  case 2:
+                    System.out.println(\"test2\");
+                    break;
+                  default:
+                    System.out.println(\"test\");
+                }
+              }
+            }",
+            "foo.java",
+            |metric| {
+                insta::assert_json_snapshot!(
+                    metric.cognitive,
+                    @r###"
+                    {
+                      "sum": 1.0,
+                      "average": 1.0,
+                      "min": 0.0,
+                      "max": 1.0
+                    }"###
+                );
+            },
+        );
+    }
+
+    #[test]
+    fn java_switch_expression() {
+        check_metrics::<JavaParser>(
+            "class X {
+              public static void print(boolean a, boolean b, boolean c, boolean d){
+                switch(expr){ // +1
+                  case 1 -> System.out.println(\"test1\");
+                  case 2 -> System.out.println(\"test2\");
+                  default -> System.out.println(\"test\");
+                }
+              }
+            }",
+            "foo.java",
+            |metric| {
+                insta::assert_json_snapshot!(
+                    metric.cognitive,
+                    @r###"
+                    {
+                      "sum": 1.0,
+                      "average": 1.0,
+                      "min": 0.0,
+                      "max": 1.0
+                    }"###
+                );
+            },
+        );
+    }
+
+    #[test]
+    fn java_not_booleans() {
+        check_metrics::<JavaParser>(
+            "class X {
+              public static void print(boolean a, boolean b, boolean c, boolean d){
+                if (a && !(b && c)) { // +1 (+1 &&, +1 &&)
```

Comment:
We can use this notation, so it is clearer why the value is 3: https://github.com/mozilla/rust-code-analysis/blob/4167e9ba17641816021c839fd1e45f3666714592/src/metrics/cognitive.rs#L951

Reply:
Sorry, @Luni-4 I miss understood what you were asking for so I changed this incorrectly. That should be correct comment now.

Reply:
@Luni-4 looks like the checks timed out waiting for an agent